api/service/transaction.ts

```diff
@@ -1,21 +1,25 @@
-import {IAsset, TransactionApi } from 'shared/model/transaction';
+import {IAsset, Transaction } from 'shared/model/transaction';
 import TransactionRepository from 'shared/repository/transaction/memory';
-import Response from 'shared/model/response';
-
-import { resGetTransactions } from 'api/controller/transaction/types';
+import ResponseEntity from 'shared/model/response';
 
 interface ITransactionService {
     createTransaction(data: any);
-    getMany(limit: number, offset: number, sort?: string, type?: number): Response<resGetTransactions>;
+    getMany(limit: number, offset: number, sort?: string, type?: number): ResponseEntity<Array<Transaction<IAsset>>>;
+    getOne(data: string): ResponseEntity<Transaction<IAsset>>;
 }
 
-class TransactionService implements ITransactionService{
+class TransactionService implements ITransactionService {
 
     createTransaction(data: any) {}
 
-    getMany(limit: number, offset: number, sort?: string, type?: number): Response<resGetTransactions> {
-        const trs = TransactionRepository.getMany(limit, offset, sort, type);
-        return new Response<resGetTransactions>({ data: trs });
+    getMany(limit: number, offset: number, sort?: string, type?: number): ResponseEntity<Array<Transaction<IAsset>>> {
```

Comment:
i think, we need to use `TransactionModel` instead of `Transaction`. Because you will have `null` for such methods as `getCopy`.